### PR TITLE
Fix machine data class

### PR DESCRIPTION
**Pull request by YujingSong**
* Move changeState() and setActiveJob() from MachineShopSimulator to Machine.
* Change machines constructor to take machines' number so that machine know its number, and we can use the array of machines less.  
* Better encapsulate jobQ by deleting the getJobQ(), instead, we implement putJobOnQ() 

Close Issue #1.

### CODE REVIEW

**Comment by miyat001** [comment]
It is great idea that move the change-state function to another class.  Especially, your team creates new value that is replacement of long if statement. It makes your code to clean and beautiful. 
The putJobOnQ is small change, but it is good idea. This change fixed the complicate code.

File: src/applications/Machine.java [hunk 1]
```diff
@@ -9,16 +9,72 @@
...
+    public Job changeState(int timeNow) {// Task on theMachine has finished,
+
+        // schedule next one.
+        Job lastJob = getActiveJob();
+        setActiveJob(timeNow);
+
+        return lastJob;
+    }
+
+    public void putJobOnQ(Job job){
```
Comment (globbycodes) [inline src/applications/Machine.java:34]: Good message chain fix.

File: src/applications/Machine.java [hunk 2]
```diff
@@ -9,16 +9,72 @@
...
+    }
+
+    /*
+    If activeJob==null: sets activeJob to the next job on the Queue, unless the jobQ is empty then we change the "state"
+                        of the machine to be idling
+    Else: set the active job to be null, signifying that the machine is in its "change over state"
+
+    Also updates all relevant numbers/fields
+     */
+    private void setActiveJob(int timeNow){
```
Comment (globbycodes) [inline src/applications/Machine.java:49]: Good idea moving logic to here. It definitely makes the MachineShopSimulator more readable!

File: src/applications/MachineShopSimulator.java [hunk 3]
```diff
@@ -32,49 +32,16 @@ static boolean moveToNextMachine(Job theJob, SimulationResults simulationResults
                 // get machine for next task
             int p = ((Task) theJob.getTaskQ().getFrontElement()).getMachine();
             // put on machine p's wait queue
-            machine[p].getJobQ().put(theJob);
+            machine[p].putJobOnQ(theJob);
             theJob.setArrivalTime(timeNow);
             // if p idle, schedule immediately
             if (eList.nextEventTime(p) == largeTime) {// machine is idle
-                changeState(p);
+                machine[p].changeState(timeNow);
```
Comment (globbycodes) [inline src/applications/MachineShopSimulator.java:39]: Now this class looks cleaner and more readable. It definitely makes more sense to have changeState() method in the Machine class.

**Comment by globbycodes** [comment]
Overall everything looks great. The pull request is well documented.

**Review by globbycodes — APPROVED** [review]

File: src/applications/Machine.java (hunk 1 above)
Comment (stev0531) [reply src/applications/Machine.java:34]: 👍

**Review by stev0531 — APPROVED** [review]
I think this was done really well. How often you committed helped me understand the reasons for the changes you were making and what those changes where doing.

File: src/applications/Machine.java (hunk 2 above)
Comment (CMenne) [reply src/applications/Machine.java:49]: I also really like this idea of moving this logic into the Machine class. By moving it here you fixed both a Data Class and Message Chain smell.

**Review by Xaivteev — APPROVED** [review]
Good changes. Makes MachineShopSimulator more readable, and makes Machine have functionality beyond simply holding data.

**Review by CMenne — APPROVED** [review]
I think the pull request is well documented with the changes that you've made. Your commit messages are informative. The refactoring you've done to MachineShopSimulator and Machine makes the code much more readable. You've fixed the Data Class smell of the Machine class and as a result addressed a Message Chain smell. Very good job. The only thing I recommend is removing the extra files from the pull request.

**Comment by miyat001** [comment]
Your explanation about issue is very clear. You explain your refactoring plan and reason. Your pull request is easy to understand what you changed in your code. The reason that why you changed the code is explained in issue section, so your pull request has enough information about your change. I cannot find commit message from your commit. Did you write commit message?

Your refactoring is very clear and complete. You create new function in Machine for reduce the lines and connection between code. 
It is great idea that move the change-state function to another class. I think this is best way to solve the bad code smell. Especially, your team creates new values (hasNoJobs, setActiveJob, getChangeTime)
that replace the long if statement. It increase the controlability and makes your code short.
The putJobOnQ is small change, but it is good idea. This change fixed the complicate code.
Moving changestate function to MachineShopSimulator is good idea. It reduces the line of code.

**Review by miyat001 — APPROVED** [review]

**Pull request merged**